api/src/groups/test_groups.py

```diff
@@ -5,993 +5,816 @@
     assert resp.status_code == 404
 
-
+#
 @pytest.mark.smoke
-def test_add_group_images_already_exist(auth_client):
-    # GIVEN a set of groups, images, and group_image relationships
-    count = random.randint(3, 6)
-    create_multiple_groups(auth_client.sqla, count)
-    create_test_images(auth_client.sqla)
-    create_images_groups(auth_client.sqla)
-
-    group_images = auth_client.sqla.query(ImageGroup).all()
-
-    # WHEN existing group_image relationships are requested to be created
-    for group_image in group_images:
-        resp = auth_client.post(
-            url_for('groups.add_group_images', group_id=group_image.group_id, image_id=group_image.image_id))
+def test_update_manager(auth_client):
+    # GIVEN an empty database
+    # WHEN we add in managers
+    create_multiple_groups(auth_client.sqla,1)
+    create_multiple_manager_types(auth_client.sqla,1)
+    create_multiple_managers(auth_client.sqla, 0.75)
+    # WHEN we get the first manager
+    first_manager = auth_client.sqla.query(Manager).first()
+    # WHEN we update an existing manager
+    resp = auth_client.patch(
+            url_for('groups.read_one_manager', group_id =first_manager.group_id, person_id =first_manager.person_id)
+            , json={'active': False})
+    # THEN we assume the correct status code
+    assert resp.status_code == 200
 
-        # THEN expect the request to be unprocessable
-        assert resp.status_code == 422
+# 
+# @pytest.mark.smoke
+def test_update_manager_invalid(auth_client):
+    # GIVEN an empty database
+    # WHEN we add in managers
+    create_multiple_groups(auth_client.sqla,1)
+    create_multiple_manager_types(auth_client.sqla,1)
+    create_multiple_managers(auth_client.sqla, 0.75)
+    # WHEN we get the first manager
+    first_manager = auth_client.sqla.query(Manager).first()
+    # THEN we update an invalid manager
+    resp = auth_client.patch(
+            url_for('groups.read_one_manager', group_id =999, person_id =first_manager.person_id)
+            , json={'active': False})
+    assert resp.status_code == 404
 
 
 @pytest.mark.smoke
-def test_delete_group_image(auth_client):
-    # GIVEN a set of groups, images, and group_image relationships
-    count = random.randint(3, 6)
-    create_multiple_groups(auth_client.sqla, count)
-    create_test_images(auth_client.sqla)
-    create_images_groups(auth_client.sqla)
-
-    valid_group_image = auth_client.sqla.query(ImageGroup).first()
-
-    # WHEN the group_image relationships are requested to be deleted
-    resp = auth_client.delete(
-        url_for('groups.delete_group_image', group_id=valid_group_image.group_id, image_id=valid_group_image.image_id))
+def test_delete_manager(auth_client):
+    # GIVEN an empty database
+    # WHEN we add in managers
+    create_multiple_groups(auth_client.sqla,1)
+    create_multiple_manager_types(auth_client.sqla,1)
+    create_multiple_managers(auth_client.sqla, 0.75)
+    # WHEN we get the first manager
```

Comment:
The `GIVEN`, `WHEN`, `THEN` pattern is supposed to read like a normal sentence. Normally when there are multiple `WHEN`s that add to one another, I write `WHEN` for the  first one then `AND` for all subsequent. So it would end up `GIVEN an empty database`, `WHEN we add managers`, `AND we get the first manager`, ... `THEN ...`.